model/metrics/clr-metrics.yaml

```diff
@@ -0,0 +1,167 @@
+groups:
+  - id: metric.clr.gc.collections.count
+    type: metric
+    metric_name: clr.gc.collections.count
+    brief: "Number of garbage collections that have occurred since the process has started."
+    instrument: counter
+    unit: "{collection}"
+    stability: experimental
+    attributes:
+      - ref: clr.gc.generation
+        requirement_level: required
+
+  - id: metric.clr.gc.objects.size
+    type: metric
+    metric_name: clr.gc.objects.size
+    brief: "The number of bytes currently allocated on the managed GC heap. Fragmentation and other GC committed memory pools are excluded."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.allocations.size
+    type: metric
+    metric_name: clr.gc.allocations.size
+    brief: >
+      The number of bytes allocated on the managed GC heap since the process has
+      started. The returned value does not include any native allocations.
+    instrument: counter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.committed_memory.size
+    type: metric
+    metric_name: clr.gc.committed_memory.size
+    brief: >
+      The amount of committed virtual memory for the managed GC heap, as
+      observed during the latest garbage collection.
+    note: >
+      Committed virtual memory may be larger than the heap size because it
+      includes both memory for storing existing objects (the heap size) and some
+      extra memory that is ready to handle newly allocated objects in the
+      future.
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.heap.size
+    type: metric
+    metric_name: clr.gc.heap.size
+    brief: >
+      The heap size (including fragmentation), as observed during the latest
+      garbage collection.
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    attributes:
+      - ref: clr.gc.generation
+        requirement_level: required
+
+  - id: metric.clr.gc.heap.fragmentation.size
+    type: metric
+    metric_name: clr.gc.heap.fragmentation.size
```

Comment:
This is the prior art for naming of fragmented bytes in .NET that I could find:
BCL API: [FragmentationBeforeBytes](https://learn.microsoft.com/en-us/dotnet/api/system.gcgenerationinfo.fragmentationbeforebytes?view=net-8.0) [FragmentationAfterBytes](https://learn.microsoft.com/en-us/dotnet/api/system.gcgenerationinfo.fragmentationafterbytes?view=net-8.0)
EventCounter: [gc-fragmentation](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/available-counters#systemruntime-counters)
OTel runtime instrumentation: [process.runtime.dotnet.gc.heap.fragmentation.size](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs#L116)

Although not identical, they all seem to agree on "fragmentation" as the term rather than "fragmented".

Reply:
left another comment suggesting `dotnet.heap.fragmentaion`